sympy/solvers/polysys.py

```diff
@@ -83,9 +80,17 @@ def solve_biquadratic(f, g, opt):
     x, y = opt.gens
 
     p = Poly(p, x, expand=False)
-    q = q.ltrim(-1)
+    p_roots = roots(p)
+    k, m = list(zip(*p_roots.items()))
+    if sum(m) == 1 and y not in k[0].free_symbols:
+        # this is not a zero-dimensional system
```

Comment:
I cannot think of a way how this could be derived from p_roots. Consider `(x**2 + y**2 - 1, x**2 - x + y**2 - 1)`.